### PR TITLE
Extra bin-dir lookup feature

### PR DESCRIPTION
**Pull request by greew**
| Q             | A
| ------------- | ---
| Branch        | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Documented?   | no
| Fixed tickets | -

The Composer bin-dir path is usually changed by setting the composer config `bin-dir`, but another common way of doing it is, if the vendor directory itself is moved by changing the `vendor-dir` configuration, by which the bin dir is moved also.

### CODE REVIEW

**Comment by veewee** [comment]
Thanks for the PR. It looks good, but before I can merge, could you please add a test-case covering this in https://github.com/phpro/grumphp/blob/master/spec/Util/ComposerFileSpec.php ?

File: src/Util/ComposerFile.php [hunk 1]
```diff
@@ -29,6 +29,12 @@ public function getBinDir(): string
         if (null !== $binDir) {
             return (string) $binDir;
         }
+        
+        $binDir = $this->configuration['config']['vendor-dir'] ?? null;
```
Comment (veewee) [inline src/Util/ComposerFile.php:33]: `$vendorDir` feels more appropriate.

**Comment by greew** [comment]
@veewee Done! :) 

Let me know if you have other requests :)

**Comment by veewee** [comment]
Thanks, that looks nice @greew .
However, now you introduced a second test class for a single unit. There was already a spec test class (the one I linked to in previous comment.)

Would you mind adding to the spec test instead? 
Another option is to move the spec cases to the phpunit testcase.

**Comment by greew** [comment]
I can do either? Which would you prefer?

(And yes, I saw your path to the file, but has never worked with PhpSpec, so didn't grasp it correctly - just thought you wrote the wrong path - sorry about that :smile: )

**Comment by veewee** [comment]
No problem 
Maybe best to move to phpunit then.
We moved away from phpspec at Phpro anyways, so that would make sense.

**Comment by greew** [comment]
There you go! All tests in the specified PhpSpec file has been moved to PhpUnit and the PhpSpec file has been deleted :)

**Comment by veewee** [comment]
Thanks!

**Pull request merged**